File: casino_test/internal/model/cascade.go
package model

type CascadeSpin struct {
	Bet int
}

// Position представляет координаты ячейки на доске
type Position struct {
	Row int
	Col int
}

// ClusterInfo содержит информацию о найденном кластере
type ClusterInfo struct {
	Symbol     int        // Символ кластера
	Cells      []Position // Позиции ячеек в кластере
	Count      int        // Количество ячеек в кластере
	Payout     int        // Выигрыш за этот кластер в деньгах
	Multiplier int        // С каким итоговым множителем ушёл кластер
}

// CascadeStep представляет один шаг каскада
type CascadeStep struct {
	CascadeIndex int           // Номер каскада (0 - первый, 1 - второй и т.д.)
	Clusters     []ClusterInfo // Информация по всем кластерам на этом шаге
	NewSymbols   []struct {
		Position
		Symbol int
	} // Какие символы упали и куда
}

// CascadeSpinResult представляет результат спина с каскадами
type CascadeSpinResult struct {
	Board            [7][7]int     // Итоговая доска после всех каскадов
	Cascades         []CascadeStep // Все шаги обновления доски
	TotalPayout      int           // Выигрыш за весь спин в деньгах
	Balance          int           // Баланс после спина в деньгах
	ScatterCount     int           // Количество бонусов, выпавших за спин
	AwardedFreeSpins int           // Количество начисленных фриспинов
	FreeSpinsLeft    int           // Остаток фриспинов после спина
	InFreeSpin       bool          // Находится ли игрок в режиме фриспинов
}

// CascadeData содержит информацию о балансе и количестве фриспинов игрока
type CascadeData struct {
	Balance       int // Теперь экспортировано (большая буква)
	FreeSpinCount int // Теперь экспортировано
}

File: ruletka/src/shared/types/cascade.ts
// Cascade Game Types

export enum CascadeSymbolType {
  EMPTY = -1,
  SYMBOL_0 = 0,
  SYMBOL_1 = 1,
  SYMBOL_2 = 2,
  SYMBOL_3 = 3,
  SYMBOL_4 = 4,
  SYMBOL_5 = 5,
  SYMBOL_6 = 6,
  SCATTER = 7,
}

export interface CascadeCell {
  symbol: number; // -1 = пусто, 0-6 = обычные, 7 = скаттер
  row: number;
  col: number;
  id: string;
  isExploding?: boolean;
  isFalling?: boolean;
}

export interface CascadeGameState {
  board: number[][]; // 7x7 доска
  balance: number;
  bet: number;
  isSpinning: boolean;
  isResolving: boolean; // Идет разрешение каскадов
  isBonusGame: boolean;
  freeSpinsLeft: number;
  lastWin: number;
  totalWin: number;
  currentCascadeIndex: number; // Текущий шаг каскада для анимации
  cascades: any[]; // Все шаги каскада
  scatterCount: number;
  awardedFreeSpins: number;
  inFreeSpin: boolean;
}

export interface CascadeCluster {
  symbol: number;
  cells: { row: number; col: number }[];
  count: number;
  payout: number;
  multiplier: number;
}

